### ISSUE TITLE
"aperture" has another simpler writting

### ISSUE THREAD

**Issue by yinsang** [30-seconds/30-seconds-of-code#1191]
```js
const aperture = (n, arr) =>
  n > arr.length
    ? []
    : arr.slice(n - 1).map((v, i) => arr.slice(i, i + n));
```
better than 
```js
const aperture = (n, arr) =>
  n > arr.length
    ? []
    : arr.slice(n - 1).map((v, i) => [...arr.slice(i, i + n - 1), v]);
```

**Comment by Chalarangelo** [comment]
Thanks for the suggestion. This looks promising. We should verify if there are any edge cases and proceed with the update. Could you open a PR for it? 😄

**Comment by yinsang** [comment]
As the project hasn't UnitTest, how contributor can test all the cases and ensure the stability of the code changing?
I am afraid of making some mistake unexpectedly.

**Comment by Chalarangelo** [comment]
@yinsang We used to have unit testing, but maintaing the tests was even more work than the code itself. To minimize contributor friction, we opted for manual testing of all the cases we can find. Contributors are encouraged to test snippets themselves and report any changes if they can.

For this specific snippet, I do not see that many things that can go wrong. Just check a few arrays with different data types and lengths, a few values for `n` and I think we should be good to go. :wink:

**Comment by yinsang** [comment]
Thanks.Refresh the concept of mine about program's UnitTest.
Indeed,Rate of flow and contrubution is open Source's soul.
I had converted *.md file to **.test.js to find bad cases of the project in the past few days.
It's actually much work, too many warnings and errors.  :(
I will keep the UnitTest in my local files and push a PR only with the *.md files some time.

**Comment by Chalarangelo** [comment]
@yinsang Cool, mind if I assign you the task then?

**Comment by yinsang** [comment]
> @yinsang Cool, mind if I assign you the task then?

Come on please! 
I just pulled a request to you.

**Comment by Chalarangelo** [comment]
Fixed in #1196
